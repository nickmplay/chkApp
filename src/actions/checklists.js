//checklist action generators

//add newChecklist object for dispatch
export const addChecklist = (newChecklist) => ({
  type: 'ADD_CHECKLIST',
  newChecklist
});

//add removeChecklist object for dispatch
export const removeChecklist = (checklistToRemoveId) => ({
  type: 'REMOVE_CHECKLIST',
  checklistToRemoveId
});