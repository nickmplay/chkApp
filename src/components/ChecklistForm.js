import React from 'react';

export class ChecklistForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      command: props.command,
      name: props.checklist ? props.checklist.name : '',
      items: props.checklist ? props.checklist.items.join(',') : ''
    };
  }
  onNameChange = (e) => {
    const name = e.target.value;
    this.setState(() => ({ name }));
  };
  onItemChange = (e) => {
    const newItem = e.target.value;
    this.setState(() => ({ items: newItem }));
  };

  onSubmit = (e) => {
    e.preventDefault();

    //form validation

    this.props.onSubmit({
      name: this.state.name,
      items: this.state.items.split(',')
    });
  };

  render() {
    return (
      <div className="content-container">
        <form className="form" onSubmit={this.onSubmit}>
          <input
            type='text'
            placeholder='Checklist Name'
            value={this.state.name}
            onChange={this.onNameChange}
          />
          <textarea
            type='text'
            placeholder='Add Items, seperating each with a comma. Eg Item 1, Item 2, Item 3'
            value={this.state.items}
            onChange={this.onItemChange}
          />
          <button className="button">{this.state.command} Checklist</button>
        </form>
      </div>
    );
  }
}