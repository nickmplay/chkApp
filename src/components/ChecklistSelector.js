import React from 'react';
import { Link } from 'react-router-dom';

export const ChecklistSelector = (props) => {
  return (
    <div>
      {props.checklists.map(checklist => {
        return (
          <div className="content-container list-item" key={checklist.id}>
            <Link to={`/view/${checklist.id}`}>
              <p  key={checklist.id + 1}>{checklist.name}</p>
            </Link>
            <Link key={checklist.id + 2} to={`/edit/${checklist.id}`}>Edit</Link>
          </div>
        );
      })}
    </div>);
};