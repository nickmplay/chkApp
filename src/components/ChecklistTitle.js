import React from 'react';
import { countChecklists } from '../selectors/checklist';

export const ChecklistTitle = (props) => {
  const checklistsWord = props.checklists.length === 1 ? 'checklist' : 'checklists';

  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">You have <span>{props.checklists.length}</span> {checklistsWord} to use:</h1>
        {/* <div className="page-header__actions">
          <Link className="button" to="/create">Add Expense</Link>
        </div> */}
      </div>
    </div>
  );
};

export default ChecklistTitle;
