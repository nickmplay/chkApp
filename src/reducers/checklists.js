//checklist reducer function

const defaultChecklistState = [];

export default (state = defaultChecklistState, action) => {
  switch (action.type) {
    case 'ADD_CHECKLIST':
      return [...state, action.newChecklist];
    case 'REMOVE_CHECKLIST':
      return state.filter(({ id }) => id !== action.checklistToRemoveId);
    case 'EDIT_CHECKLIST':
      return state.map((e) => {
        if (e.id === action.checklistUpdates.id) {
          return {
            id: e.id,
            items: action.checklistUpdates.items
          };
        } else {
          return e; 
        }
      });
    default:
      return state;
  }
};
