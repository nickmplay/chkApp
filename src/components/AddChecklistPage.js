import React from 'react';
import { connect } from 'react-redux';
import { addChecklist } from '../actions/checklists';
import { startAddChecklist } from '../actions/firebase';
import { ChecklistForm } from './ChecklistForm';

export class AddChecklistPage extends React.Component {
  onSubmit = ({name, items}) => {
    this.props.startAddChecklist({
      name,
      items
    });
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Checklist</h1>
          </div>
        </div>
        <ChecklistForm 
          command="Add"
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddChecklist: (checklist) => dispatch(startAddChecklist(checklist))
});

export default connect(undefined, mapDispatchToProps)(AddChecklistPage);
