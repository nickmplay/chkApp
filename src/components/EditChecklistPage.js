import React from 'react';
import { connect } from 'react-redux';
import { editChecklist } from '../actions/checklists';
import { ChecklistForm } from './ChecklistForm';

export class EditChecklistPage extends React.Component {
  onSubmit = (checklist) => {
    this.props.editChecklist(checklist);
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit {this.props.checklist.name} checklist</h1>
          </div>
        </div>
        <ChecklistForm 
          command="Edit"
          checklist={this.props.checklist}
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  checklist: state.checklists.find( e => e.id.toString() === props.match.params.id)
});

const mapDispatchToProps = (dispatch) => ({
  editChecklist: (checklist) => dispatch(editChecklist(checklist))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditChecklistPage);
