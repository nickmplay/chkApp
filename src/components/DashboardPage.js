import React from 'react';
import { connect } from 'react-redux';
import { ChecklistTitle } from './ChecklistTitle';

export const DashboardPage = (props) => (
  <div>
    <ChecklistTitle checklists={props.checklists} />
    {props.checklists.map( checklist => {
      return <p className="content-container" key={checklist.id}>{checklist.name}</p>;
    })}
  </div>
);

const mapStateToProps = (state, props) => ({
  checklists: state.checklists
});

export default connect(mapStateToProps)(DashboardPage);
