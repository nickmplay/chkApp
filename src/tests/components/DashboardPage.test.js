import React from 'react';
import { shallow } from 'enzyme';
import { DashboardPage } from '../../components/DashboardPage';
import checklists from '../fixtures/checklists';

test('should render DashboardPage correctly with data', () => {
  const wrapper = shallow(<DashboardPage checklists={checklists} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render DashboardPage correctly without data', () => {
  const wrapper = shallow(<DashboardPage checklists={[]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render DashboardPage correctly with only one', () => {
  const wrapper = shallow(<DashboardPage checklists={[checklists[0]]} />);
  expect(wrapper).toMatchSnapshot();
});