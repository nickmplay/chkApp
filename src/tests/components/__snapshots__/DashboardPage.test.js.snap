// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DashboardPage correctly with data 1`] = `
<div>
  <ChecklistTitle
    checklists={
      Array [
        Object {
          "id": 1,
          "items": Array [
            "lock back door",
            "lock front door",
            "turn off lights",
          ],
          "name": "bed time",
        },
        Object {
          "id": 2,
          "items": Array [
            "rucksack 1",
            "rucksack 2",
          ],
          "name": "car",
        },
        Object {
          "id": 3,
          "items": Array [
            "fridge light",
            "turn on urn",
          ],
          "name": "friday",
        },
      ]
    }
  />
  <p
    className="content-container"
    key="1"
  >
    bed time
  </p>
  <p
    className="content-container"
    key="2"
  >
    car
  </p>
  <p
    className="content-container"
    key="3"
  >
    friday
  </p>
</div>
`;

exports[`should render DashboardPage correctly with only one 1`] = `
<div>
  <ChecklistTitle
    checklists={
      Array [
        Object {
          "id": 1,
          "items": Array [
            "lock back door",
            "lock front door",
            "turn off lights",
          ],
          "name": "bed time",
        },
      ]
    }
  />
  <p
    className="content-container"
    key="1"
  >
    bed time
  </p>
</div>
`;

exports[`should render DashboardPage correctly without data 1`] = `
<div>
  <ChecklistTitle
    checklists={Array []}
  />
</div>
`;
